src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/ChainFlatteningRewriter.cs

```diff
@@ -52,34 +52,72 @@ public override Expression VisitSqlRoot(SqlRootExpression expression, object con
 
         private void ProcessChainedExpression(ChainedExpression chainedExpression, List<SearchParamTableExpression> tableExpressions, int chainLevel)
         {
-            SearchParamTableExpressionQueryGenerator queryGenerator = chainedExpression.Expression.AcceptVisitor(_searchParamTableExpressionQueryGeneratorFactory, null);
+            if (!(chainedExpression.Expression is MultiaryExpression multiaryExpression && multiaryExpression.MultiaryOperation == MultiaryOperator.And))
```

Comment:
nit: I feel that the conditional checks should be reversed. Since `HandleAndExpression` is the more restrictive scenario, we should specifically check for that under the `if` condition and have the normal behavior under the `else` block.

Reply:
I put it that way in hope what codefow would pick up changes in a right way, instead of mixing them, but that didn't happened. So back to other order.